cleverhans/utils_tf.py

```diff
@@ -265,15 +267,17 @@ def model_eval(sess, x, y, predictions=None, X_test=None, Y_test=None,
             start = batch * args.batch_size
             end = min(len(X_test), start + args.batch_size)
             cur_batch_size = end - start
+            X_cur[:cur_batch_size] = X_test[start:end]
+            Y_cur[:cur_batch_size] = Y_test[start:end]
 
             # The last batch may be smaller than all others, so we need to
             # account for variable batch size here
-            feed_dict = {x: X_test[start:end], y: Y_test[start:end]}
+            feed_dict = {x: X_cur, y: Y_cur}
             if feed is not None:
                 feed_dict.update(feed)
-            cur_acc = acc_value.eval(feed_dict=feed_dict)
+            cur_corr_preds = correct_preds.eval(feed_dict=feed_dict)
 
-            accuracy += (cur_batch_size * cur_acc)
+            accuracy += cur_corr_preds[:cur_batch_size].sum()
```

Comment:
Thanks, this looks like a better implementation.
Do you know if the old version was giving inaccurate values for any of the benchmarks?

Reply:
The benchmarks might have been slightly affected. On MNIST, the test set size is 10,000 which is not divisible by the typical mini-batch size 128.